packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -148,15 +148,10 @@ rocksdb::Status ArbCore::initialize(const LoadedExecutable& executable) {
     // Use latest existing checkpoint
     ValueCache cache{1, 0};
 
-    auto message_count_result = unexpiredMessageCount();
-    if (!message_count_result.status.ok()) {
-        return message_count_result.status;
-    }
-
-    auto status = reorgToMessageCountOrBefore(
+    auto status = reorgCheckpoints(
         [&](const MachineOutput& output) {
-            return message_count_result.data >=
-                   output.fully_processed_inbox.count;
+            return timed_sideload_cache.expiredTimestamp() >=
```

Comment:
I think we should store `expiredTimestamp()` outside the call since otherwise the check could return differently for the same input if separated in time which could mess up `reorgCheckpoints`

Reply:
refactored call into `reorgToTimestampOrBefore` which has a side effect of doing so.